### DISCUSSION TITLE
Where do I now import createMetadataMap? -- upgraded from 2.1.1

### DISCUSSION THREAD

**Discussion by johnpankowicz** [nartc/mapper#459] [Q&A] [answered]
I just updated an Angular project. While doing that I updated my automapper npm packages from ^2.1.1 to ^8.3.2. I now get an error on my import statement:

`import { createMetadataMap, pojos } from '@automapper/pojos';`

`@automapper/pojos"' has no exported member 'createMetadataMap'`

Where do I import this from now? Also, will I need to do a lot of other refactoring when upgrading from 2.1.1. to 8.3.2?

**Comment by nartc** [comment, marked as answer]
Hi @johnpankowicz, if you use `pojos`, then probably more migrations to do than usual. I'd recommend checking out the migration guide (https://automapperts.netlify.app/docs/getting-started/migrate-to-automapper-v8#automapperpojos). Hit me up if you have more questions

**Reply by johnpankowicz** [reply]
Thanks. I fixed the createMetadataMap calls. 

I also modified my createMapper() call as shown for pojos

from
```
const mapper = createMapper({
    pluginInitializer: pojos, // no ()
});
```
to:
```
const mapper = createMapper({
    strategyInitializer: pojos(), // invoking
});
```
But it tells me: 
```
(property) GovbodyMapper.mapper: Mapper
Type 'Mapper' is missing the following properties from type 'Mapper': name, plugin, addTypeConverter, createMap, and 2 more.ts(2740)
```
Could I have a problem with mismatched npm packages? package.json shows I have @automapper/pojos": "^8.3.2" and @automapper/types": "^6.3.1"

**Reply by nartc** [reply]
You can just uninstall @automapper/types
